# input foramt
# 1 for 422 input, 0: for 420 input
input_format:             1

# LDC LUT definition
lut_file_path:            $(find ti_vl)/ldc_lut/zero_768x384_LUT.bin

# location of DL model
dl_model_path:            $(env TI_DATA_PATH)/visual_localization/dkaze_768x384_inference_tvm-dlr

# location of top-view image
top_view_img_path:        $(env TI_DATA_PATH)/visual_localization/map/topview_gt_0532.yuv

# location of voxel info
input_voxel_info_path:    $(env TI_DATA_PATH)/visual_localization/map/map_voxel.bin

# location of map feature points
input_map_feat_pt_path:   $(env TI_DATA_PATH)/visual_localization/map/map_feat.bin

# location of map feature descriptors
input_map_feat_desc_path: $(env TI_DATA_PATH)/visual_localization/map/map_desc.bin

# location of input upsampling weight
input_upsample_wt_path:  $(env TI_DATA_PATH)/visual_localization/map/weights.bin

# location of input upsampling bias
input_upsample_bias_path: $(env TI_DATA_PATH)/visual_localization/map/bias.bin

# input data width in pixels
width:                    768

# input data height in lines
height:                   384

# output width
out_width:                2048

# output height
out_height:               1024

# Score threshold for picking good points
score_th:                 128

# map related parameters
# max number of frame features
max_frame_feat:           1000

# number of voxels
num_voxels:               111556

# number of features in a map
num_map_feat:             13718

# initial pose estimate used in campling the map data 
init_pose_est:            [-156.546, 0.0, -132.51]

# should be consistant with scale of filter coefficients passed through file input_upsampling_weight
filter_scale_pw2:          8

# should be consistant with scale of descriptors present in input_map_feat_desc_file
hi_res_desc_scale_pw2:     0

# interactive mode
is_interactive:            0

# To enable showing output on display when display_option = 1
display_option:            1

# pipeline depth
pipeline_depth:            4

# Flag to indicate if the graph should be exported
#  0 - disable
#  1 - enable
exportGraph:               0

# Flag to indicate if the performance data should be exported
# to a file
#  0 - disable
#  1 - enable
exportPerfStats:           0

# Real-time logging enable.
#  0 - disable
#  1 - enable
rtLogEnable:               0

# log level for the app
# 0 - DEBUG: Used to show general debug messages
# 1 - INFO: Used to show run-time processing debug
# 2 - WARN: Used to warning developers of possible issues
# 3 - ERROR: Used for most errors
log_level:                 1
